File: src/main/java/it/xpug/kata/birthday_greetings/XDate.kt
package it.xpug.kata.birthday_greetings

import java.time.LocalDate
import java.time.format.DateTimeFormatter

data class XDate(val date: LocalDate) {

    constructor(yyyyMMdd: String) : this(LocalDate.parse(yyyyMMdd, FORMATTER))

    val day: Int
        get() = date.dayOfMonth

    val month: Int
        get() = date.monthValue

    fun isSameDay(anotherDate: XDate): Boolean {
        return anotherDate.day == this.day && anotherDate.month == this.month
    }

    companion object {
        private val FORMATTER: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy/MM/dd")
    }
}

File: csharp/BirthdayGreetings/XDate.cs
using System;
using System.Globalization;

namespace BirthdayGreetings
{
    public class XDate
    {
        private DateTime date;

        public XDate(string yyyyMMdd)
        {
            date = DateTime.ParseExact(yyyyMMdd, "yyyy/MM/dd", CultureInfo.InvariantCulture);
        }

        public int Day => date.Day;
        public int Month => date.Month;

        public bool IsSameDay(XDate anotherDate)
        {
            return anotherDate.Day == this.Day && anotherDate.Month == this.Month;
        }

        public override string ToString()
        {
            return date.ToString("yyyy-MM-dd");
        }
    }
} 